Lib9c/Action/Stake.cs

```diff
@@ -190,35 +191,46 @@ private static IWorld ContractNewStake(
                     .TransferAsset(context, context.Signer, stakeStateAddr, additionalBalance)
                     .MintAsset(context, stakeStateAddr, gg);
 
-                try
+                var guildRepository = new GuildRepository(state, context);
+                if (guildRepository.TryGetGuildParticipant(agentAddress, out var guildParticipant))
                 {
-                    var guildRepository = new GuildRepository(state, context);
-                    var guildParticipant = guildRepository.GetGuildParticipant(context.Signer);
                     var guild = guildRepository.GetGuild(guildParticipant.GuildAddress);
                     guildParticipant.Delegate(guild, gg, height);
                     state = guildRepository.World;
                 }
-                catch (FailedLoadStateException)
+                else
                 {
+                    state = state
+                        .TransferAsset(context, stakeStateAddr, Addresses.NonValidatorDelegatee, gg);
                 }
-                
+
             }
             else if (additionalBalance.Sign < 0)
             {
                 var gg = GetGuildCoinFromNCG(-additionalBalance);
 
-                try
+                var guildRepository = new GuildRepository(state, context);
+                if (guildRepository.TryGetGuildParticipant(agentAddress, out var guildParticipant))
                 {
-                    var guildRepository = new GuildRepository(state, context);
-                    var guildParticipant = guildRepository.GetGuildParticipant(context.Signer);
                     var guild = guildRepository.GetGuild(guildParticipant.GuildAddress);
                     var share = guild.ShareFromFAV(gg);
                     guildParticipant.Undelegate(guild, share, height);
                     state = guildRepository.World;
                 }
-                catch (FailedLoadStateException)
+                else
                 {
-                    // TODO: Create self unbonding
+                    var delegateeAddress = Addresses.NonValidatorDelegatee;
+                    var delegatorAddress = context.Signer;
+                    var repository = new ValidatorRepository(state, context);
+                    var unbondLockInAddress = DelegationAddress.UnbondLockInAddress(delegateeAddress, delegatorAddress);
+                    var unbondLockIn = new UnbondLockIn(
+                        unbondLockInAddress, 1, delegateeAddress, delegatorAddress, null);
```

Comment:
> Any reason for limit `maxEntries` as 1? Limiting this will block users to unstake repeatedly.

I used a value of 1 because I didn't know what value would be appropriate. If you can suggest an appropriate value, I'll use it.

Reply:
Committed https://github.com/planetarium/lib9c/pull/2976/commits/afb94731a485a1d7fbfa2a94031144f29480e30e